src/middlewared/middlewared/plugins/pool_/snapshot_count.py

```diff
@@ -0,0 +1,23 @@
+from pathlib import Path
+
+from middlewared.plugins.zfs_.utils import ZFSCTL
+from middlewared.schema import accepts, returns, Int, Str
+from middlewared.service import Service
+
+
+class PoolDatasetService(Service):
+
+    class Config:
+        namespace = "pool.dataset"
+
+    @accepts(Str("dataset"))
+    @returns(Int())
+    def snapshot_count(self, dataset):
+        mountpoint = Path("/mnt") / dataset
+        zfs_dir = mountpoint / ".zfs/snapshot"
+        if zfs_dir.is_dir():
+            stat = zfs_dir.stat()
+            if stat.st_ino == ZFSCTL.INO_SNAPDIR:
+                return stat.st_nlink - 2
```

Comment:
@anodos325 IIRC this is how it should work?

Reply:
Yeah. That should be fine.